src/llama_index_alloydb_pg/engine.py

```diff
@@ -767,6 +767,98 @@ def init_index_store_table(
             )
         )
 
+    async def _ainit_chat_store_table(
+        self,
+        table_name: str,
+        schema_name: str = "public",
+        overwrite_existing: bool = False,
+    ) -> None:
+        """
+        Create an AlloyDB table to save chat store.
+
+        Args:
+            table_name (str): The table name to store chat history.
+            schema_name (str): The schema name to store the chat store table.
+                Default: "public".
+            overwrite_existing (bool): Whether to drop existing table.
+                Default: False.
+
+        Returns:
+            None
+        """
+        if overwrite_existing:
+            async with self._pool.connect() as conn:
+                await conn.execute(
+                    text(f'DROP TABLE IF EXISTS "{schema_name}"."{table_name}"')
+                )
+                await conn.commit()
+
+        create_table_query = f"""CREATE TABLE "{schema_name}"."{table_name}"(
+            id SERIAL PRIMARY KEY,
+            key VARCHAR NOT NULL,
+            messages JSON[] NOT NULL,
+            CONSTRAINT "{table_name}_unique_key" UNIQUE (key)
```

Comment:
The key will not be unique. Each key will have mulitple rows with 1 message per row with an unique id for ordering